common/djangoapps/third_party_auth/api/views.py

```diff
@@ -249,6 +250,42 @@ def get(self, request):
         identifier = self.get_identifier_for_requested_user(request)
         return self.do_get(request, identifier)
 
+    def delete(self, request):
+        """
+        Delete given social auth record for a user.
+
+        Args:
+            request (Request): The HTTP DELETE request
+
+        Request Parameters:
+            email/username: Must provide one of 'email' or 'username'.  If both are provided,
+            the username will be ignored.
+            uid: UID of the social auth record to delete
+
+        Return:
+            JSON serialized list of the providers linked to this user after the delete operation.
+
+        """
+        identifier = self.get_identifier_for_requested_user(request)
+        uid = request.query_params.get("uid")
+        if not uid:
+            raise exceptions.ValidationError("Must provide uid")
+
+        is_unprivileged = self.is_unprivileged_query(request, identifier)
+
+        if is_unprivileged:
+            return Response(status=status.HTTP_403_FORBIDDEN)
```

Comment:
We don't have a test case for this condition, it will be a good idea to test a user who does not have permission to delete the record.

Reply:
@farhaanbukhsh Added